In [16]:
import numpy as np
import pandas as pd

In [17]:
data = pd.read_csv("data.csv")
data.head()

,SALESPERSON,Old Scheme (in thousands),New Scheme (in thousands)
0,1,57,62
1,2,103,122
2,3,59,54
3,4,75,82
4,5,84,84


1. Find the mean of old scheme and new scheme column

In [18]:
print("mean of old scheme")
data["Old Scheme (in thousands)"].mean()

mean of old scheme


68.03333333333333

In [19]:
print("mean of new scheme")
data["New Scheme (in thousands)"].mean()

mean of new scheme


72.03333333333333

2. Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs?

In [82]:
import pandas as pd
from scipy.stats import ttest_ind,ttest_1samp, wilcoxon,stats
old_scheme = data["Old Scheme (in thousands)"]
new_scheme = data["New Scheme (in thousands)"]

#t_statistic, p_value = ttest_ind(old_scheme, new_scheme)
t_statistic, p_value =stats.ttest_rel(old_scheme,new_scheme)

print(t_statistic, p_value)
print("p_value ",p_value)
print("Accepting H0")
print("there is no new scheme has significantly raised outputs")

-1.5559143823544377 0.13057553961337662
p_value  0.13057553961337662
Accepting H0
there is no new scheme has significantly raised outputs


3. What conclusion does the test (p-value) lead to? 

alpha=0.05
p-value=0.13057553961337662
Since p-value>alpha, so rejecting H0
0.13 indicates weak evidence against null hypothesis

4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:

a) The probability of a type 1 error?

Probability of Type I error = alpha = 0.05

b) What is the p- value of the hypothesis test if we test for a difference of $5000? 

In [78]:
from scipy import stats
from scipy.stats import t
from statsmodels.stats.power import ttest_power
t_statistic, p_value =stats.ttest_rel(old_scheme,new_scheme)
t_statistic
p_value


0.13057553961337662

In [23]:
df1=data["Old Scheme (in thousands)"]+5

In [22]:
df1.mean()

77.03333333333333

In [86]:
t_statistic, p_value = ttest_ind(old_scheme, 5)
t_statistic, p_value =stats.ttest_rel(old_scheme,new_scheme)
print(t_statistic, p_value)


-1.5559143823544377 0.13057553961337662


3.6719608244501054

In [29]:
print("p- value of the hypothesis for a difference of $5000",p_value)

p- value of the hypothesis for a difference of $5000 0.34773807155709624


c) Power of the test